### PR TITLE
Bug 1292576 - Generalize the carton bundle creation scripts

### PR DESCRIPTION
**Pull request by dylanwh**

### CODE REVIEW

File: bugzilla6/Dockerfile.PL [hunk 1]
```diff
@@ -1,20 +0,0 @@
...
-        expat-devel
-        gd-devel
-        httpd-devel
-        mysql-devel
-        openssl
-        openssl-devel
-        perl-core
-};
-
-build_tarball();
```
Comment (dklawren) [inline bugzilla6/Dockerfile.PL]: Why did you remove bugzilla6/Dockerfile.PL?

**Comment by dklawren** [comment]
bmo_merge fails to build for me: https://dkl.pastebin.mozilla.org/8890861

File: bugzilla6/Dockerfile.PL (hunk 1 above)
Comment (dylanwh) [inline bugzilla6/Dockerfile.PL]: because it won't build until the XSLoader problem is fixed in the cpan ecosystem -- or untill upstream bugzilla pins its version of List::MoreUtil to a version less than the current one.

File: bmo/Dockerfile.PL [hunk 2]
```diff
@@ -20,5 +15,6 @@ RUN q{
     yum clean all
 };
 
+git_clone("git://github.com/dylanwh/bmo.git");
```
Comment (dklawren) [inline bmo/Dockerfile.PL]: The final cut should come from https://github.com/mozilla-bteam/bmo instead.

**Comment by dylanwh** [comment]
bmo_merge is a WIP.  
I was starting from zero rpms and seeing what needs to exist for it to work

File: mozreview/Dockerfile.PL [hunk 3]
```diff
@@ -34,4 +33,6 @@ RUN qq{
         wget
 };
 
+git_clone('git://github.com/dylanwh/bmo.git');
```
Comment (dklawren) [inline mozreview/Dockerfile.PL]: Same here (official repo)

File: bmo/Dockerfile.PL (hunk 2 above)
Comment (dylanwh) [inline bmo/Dockerfile.PL]: Yup, but it's a chicken and egg problem. we can't push to master until the vendor tarballs are installed on the servers.

**Comment by dklawren** [comment]
Also mozreview fails to build for me as well. Something with my environment?

https://dkl.pastebin.mozilla.org/8890873

File: Makefile [hunk 4]
```diff
@@ -9,18 +11,41 @@ SCRIPTS   := $(wildcard scripts/*)
...
 build-%: %/Dockerfile %/.dockerignore $(SCRIPTS)
-	cd $* && $(DOCKER) build -m 2G -t $(IMAGE_TAG) . > build.log
-	$(DOCKER) images -q $@ > $@
+	@echo BUILD $*
+	@cd $* && $(DOCKER) build -m 2G -t $(IMAGE_TAG) . > build.log
+	@$(DOCKER) images -q $@ > $@
+
+clean-%:
+	@echo CLEAN $*
+	@echo rm -vf $*/Dockerfile $*/vendor.tar.gz $*/*.log $*/*.tmp
```
Comment (dklawren) [inline Makefile]: Should be instead:

@rm -vf $_/Dockerfile $_/vendor.tar.gz $_/_.log $_/_.tmp
